core-bundle/src/EventListener/DataContainer/RecordPreviewListener.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener\DataContainer;
+
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\CoreBundle\ServiceAnnotation\Hook;
+use Contao\DataContainer;
+use Contao\System;
+use Doctrine\DBAL\Connection;
+
+/**
+ * @internal
+ */
+class RecordPreviewListener
+{
+    private ContaoFramework $framework;
+
+    private Connection $connection;
+
+    public function __construct(ContaoFramework $framework, Connection $connection)
+    {
+        $this->framework = $framework;
+        $this->connection = $connection;
+    }
+
+    /**
+     * @Hook("loadDataContainer")
+     */
+    public function registerDeleteCallbacks(string $table): void
+    {
+        if ($GLOBALS['TL_DCA'][$table]['config']['notDeletable'] ?? false) {
+            return;
+        }
+
+        if (!isset($GLOBALS['TL_DCA'][$table]['config']['ondelete_callback'])) {
+            $GLOBALS['TL_DCA'][$table]['config']['ondelete_callback'] = [];
+        }
+
```

Comment:
```suggestion
```
This is not needed as PHP automatically sets the array when calling `$GLOBALS['TL_DCA'][$table]['config']['ondelete_callback'][] = …`